core/arch/x86/emit_utils.c

```diff
@@ -3155,6 +3155,50 @@ relink_special_ibl_xfer(dcontext_t *dcontext, int index,
     protect_generated_code(code, READONLY);
 }
 
+/*
+RAW(66) RAW(90)
+RAW(67) RAW(90)
+RAW(f2) RAW(90)
+RAW(f3) RAW(90)
+RAW(66) RAW(66) RAW(66) RAW(66) RAW(66) RAW(90)
+RAW(0f) RAW(1f) RAW(00)
+RAW(0f) RAW(1f) RAW(40) RAW(00)
+RAW(0f) RAW(1f) RAW(44) RAW(00) RAW(00)
+RAW(66) RAW(0f) RAW(1f) RAW(44) RAW(00) RAW(00)
+RAW(0f) RAW(1f) RAW(80) RAW(00) RAW(00) RAW(00) RAW(00)
+RAW(0f) RAW(1f) RAW(84) RAW(00) RAW(00) RAW(00) RAW(00) RAW(00)
+RAW(66) RAW(0f) RAW(1f) RAW(84) RAW(00) RAW(00) RAW(00) RAW(00) RAW(00)
+*/
+bool
+fill_with_nops(dr_isa_mode_t isa_mode, byte *addr, size_t size)
+{
+    if (isa_mode == DR_ISA_AMD64) {
+        /* 64 bit mode can take advantage of multibyte nops.
+         * ref. AMD Software Optimization Guide for AMD Family 15h Processors, document
+         * #47414, section 5.8 "Code Padding with Operand-Size Override and Multibyte
+         * NOP", page 94)
+         */
+        switch (size) {
```

Comment:
You mean an encode/decode test of these raw byte sequences?  Sharing them somehow with that test?  (Because SET_TO_NOPS/fill_with_nops is not an interface function exposed in the API and so can't be called from api.ir.  It could be called from unit_tests added to emit_utils.c.)

Xref INSTR_CREATE_nop2byte_reg().  You could imagine adding this support for larger nops there too, sharing the byte sequences somehow.  Though let's not pile on too much here in this PR: if @ZehMatt wants to keep it just optimizing the internal fill routine, just adding a TODO comment about sharing with INSTR_CREATE_nop2byte_reg seems fine.

Reply:
I won't mind creating a new PR after this one, so I would prefer to keep it simple when possible and then tackle the next issue which sounds like we want to share the sequence/data also with the API.

Reply:
Ok!